src/ploigos_step_runner/step_implementers/generate_metadata/git.py

```diff
@@ -107,6 +107,8 @@ def _run_step(self):
             step_result.message = 'Given directory (repo_root) is a bare Git repository'
             return step_result
 
+        # The SemanticVersion StepImplementer requires the branch name (as stored by 'pre-release' below),
```

Comment:
i woudl re-phrase this to be more postiive. something like "this step implmenter use the branch name as the pre-release value, therefor this step impleenter requires a non-detached head, if anyone has any brilliant ideas for determine if shoudl specify a pre-release value in a detached head state, pull requests welcome"

Reply:
Updated!